ideascube/serveradmin/models.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+
+class Setting(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def _get_typed(cls, namespace, key, type, default=None):
+        try:
+            setting = Setting.objects.get(namespace=namespace, key=key)
+
+        except Setting.DoesNotExist:
+            if default is not None:
+                return default
+
+            raise
+
+        else:
+            if isinstance(setting.value, type):
+                return setting.value
+
+            if default is not None:
```

Comment:
> At the time the error will raise, there will be no developer to be able to fix the situation.

There are a lot of situation when no developer is around to fix a bug :)

How the values will be set ? Is it the user (ansiblecube) directly setting the value in the db or we use API that will put a value of the right type in the db ?

If the user set directly in the db, then the db is like a config file and we need a defensive approach.

Else, there is no reason* to be more defensive than elsewhere. We (developer) write something in the db, and we can't read it back. This should not happen.

\* It may be one reason : evolution of the settings. But I think that it should be to the using code to be prepared to and to handle this case.

However, trying to have a defensive approach is still a valid argument,.I will not block this pr for this. (If, at least, there is a warning message)